test/extended/util/framework.go

```diff
@@ -2002,3 +2006,18 @@ func AllClusterVersionsAreGTE(version semver.Version, config *rest.Config) (bool
 	}
 	return true, nil
 }
+
+// GetControlPlaneTopology retrieves the cluster infrastructure TopologyMode
+func GetControlPlaneTopology(ocClient *CLI) (*configv1.TopologyMode, error) {
+	if ControlPlaneTopology == nil {
```

Comment:
I think this needs to be threadsafe/protected by a sync.Mutex as tests could end up calling this in parallel

Reply:
Added a mutex lock to the function

Reply:
the mutex already needs to be used for reading ( `if ControlPlaneTopology == nil`) as reading/writing to a variable in parallel is a datarace.

Reply:
Will the additional locking be acceptable, if we always perform it when this function is called, compared with the additional check after receiving the lock, prior to attempting to set the value?

I considered using the builtin `once`, but was not confident that that would work properly if an error was raised while attempting to get the `Infrastructure.Status.ControlPlaneTopology`.

Reply:
If we just lock in the very beginning of the function prior to reading we are good. An additional check prior to setting the value would only be needed if we used a RWLock, but I don't think we need that.

Reply:
okay, I'll make that change, thanks!

Reply:
updated